### ISSUE TITLE
Migrate this project to TypeScript

### ISSUE THREAD

**Issue by g-plane** [vuejs/vue-test-utils#1103]
### What problem does this feature solve?
TypeScript is more and more popular at present. Will the source code of this project migrate to TypeScript? This can let definition files be generated automatically.

### What does the proposed API look like?
N/A

**Comment by eddyerburgh** [comment]
Yes there is a plan to migrate to TypeScript, but I'm focussed on getting Vue Test Utils to 1.0 before migrating. If somebody else would like to migrate then I am happy to review.

Note that I think we should keep the test files as JavaScript.

**Comment by g-plane** [comment]
> Note that I think we should keep the test files as JavaScript.

Are `the test files` of this project or of users' test files?

**Comment by eddyerburgh** [comment]
The test files of this project, in the test directory

**Comment by FernandoBasso** [comment]
I would be interested in helping with this.

**Comment by lmiller1990** [comment]
As would I. I have been working with TS a lot lately, and some of Vue's pain points around TS support (which will change with 3.0) have become clear. Anything that improves TS support is a big plus from me.

Has anyone got experience moving a large scale project to TS? This seems challenging, since the Flow types might clash. I'm guessing the best way is

1. add the things required to compile ts (`yarn add typescript`). Set it up to allow you to import `.js. files. This would be the branch all TS conversion PRs target, likely `epic/typescript-migration` or something, that kicks the migration off.
2. pick a simple file. Change it to `.ts`, and make sure the tests still pass. 
3. Add types to said `.ts` file
4. Repeat

[This post](https://medium.com/entria/incremental-migration-to-typescript-on-a-flowtype-codebase-515f6490d92d) has some details on how to handle flow/ts in the same codebase.

**Comment by dsseng** [comment]
I can migrate some pieces of code to TS. @eddyerburgh may I do it?

**Comment by eddyerburgh** [comment]
Yes @sh7dm 👍 . But please keep tests in JavaScript for now

**Comment by dsseng** [comment]
Ok
